include/maliput_sparse/parser/lane.h

```diff
@@ -0,0 +1,69 @@
+// BSD 3-Clause License
+//
+// Copyright (c) 2022, Woven Planet.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include <optional>
+#include <string>
+#include <unordered_set>
+
+#include "maliput_sparse/geometry/line_string.h"
+
+namespace maliput_sparse {
+namespace parser {
+
+/// Contains the characteristics of a lane.
+struct Lane {
```

Comment:
OSM itself has a width and a set of waypoints instead of a left and right sets. This struct is not generic at a glance.

We are also missing the most complicated cases in which surface information (e.g. CRG files) are provided. There is no support for that in these structs. This point is not a strong argument yet as we are not able to support it in any part of this implementation yet.

Reply:
> This struct is not generic at a glance.

I think that these structs should contain what we can handle with the builder api because at the end of the day the backend's creator must match the information from the parser to the builder. How can we make them even more generic?

Reply:
Perhaps the linestring construction should happen outside this struct. The first time I read this PR, I made a mental model and intent to make this generic for other sparse formats, so restricting at the "parser" level the geometric model when the format is unknown it doesn't sound a good approach to me.

I may be missing context, then I think it's a matter of facing the use case in which another geometry model is required and adapting the code accordingly.